frontend/amundsen_application/static/js/ducks/tableMetadata/api/helpers.ts

```diff
@@ -32,16 +34,31 @@ export function getRelatedDashboardSlug(key: string): string {
   return encodeURIComponent(key);
 }
 
+/**
+ *
+ * @param columns
+ */
+export function parseNestedColumns(columns: TableColumn[]) {
+  columns.forEach((column) => {
+    const nestedType = parseNestedType(column.col_type);
+    if (nestedType !== null) {
+      column.children = convertNestedTypeToColumns(nestedType);
+    }
+  });
+};
+
 /**
  * Parses the response for table metadata information to create a TableMetadata object
  */
 export function getTableDataFromResponseData(
   responseData: API.TableDataAPI
 ): TableMetadata {
-  return filterFromObj(responseData.tableData, [
+  const tableData = filterFromObj(responseData.tableData, [
     'owners',
     'tags',
   ]) as TableMetadata;
+  parseNestedColumns(responseData.tableData.columns);
```

Comment:
Could we change this so we don't mutate the original array?
I would change the 'forEach' into a map and save the results.